spikeinterface/comparison/hybridspikesrecording.py

```diff
@@ -0,0 +1,45 @@
+from typing import List, Union
+import numpy as np
+from spikeinterface.core import BaseSorting, NumpySorting, WaveformExtractor, AddTemplatesRecording
+
+
+class HybridSpikesRecording(AddTemplatesRecording):
+
+	def __init__(self, wvf_extractor: WaveformExtractor, max_n_spikes: int = 1000,
+				 max_frac_spikes: float = 0.05, refractory_period: float = 1.5) -> None:
+		target_recording = wvf_extractor.recording
+		target_sorting = wvf_extractor.sorting
+		templates = wvf_extractor.get_all_templates()
+		injected_sorting = self._generate_injected_sorting(target_sorting, recording.get_num_frames(),
+						   max_n_spikes, max_frac_spikes, refractory_period)
+
+		AddTemplatesRecording.__init__(target_sorting, templates, wvf_extractor.nbefore, target_recording=target_recording)
+
+
+	# TODO: May be a better way of creating this?		
+	def _generate_injected_sorting(sorting: BaseSorting, t_max: int, max_n_spikes: int,
+								   max_frac_spikes: float, refractory_period: float) -> NumpySorting:
+		injected_spike_trains = {}
+		t_r = int(round(refractory_period * sorting.get_sampling_frequency() * 1e-3))
+
+		for unit_id in sorting.unit_ids:
```

Comment:
I would allow to select a subset of `unit_ids` (`=None` by default)

Reply:
Can't people just give `sorting.select_units(unit_ids)`?
I guess I can still add it to make it easier.

Reply:
you're passing a waveform extractor!

Reply:
Ah true ^^'
I'll add it!

Reply:
Done.